fastly/acl_entry.go

```diff
@@ -58,23 +43,14 @@ type ListACLEntriesInput struct {
 	Sort string
 }
 
-func (l *ListACLEntriesInput) formatFilters() map[string]string {
-	m := make(map[string]string)
-
-	if l.Direction != "" {
-		m["direction"] = l.Direction
-	}
-	if l.Page != 0 {
-		m["page"] = strconv.Itoa(l.Page)
-	}
-	if l.PerPage != 0 {
-		m["per_page"] = strconv.Itoa(l.PerPage)
-	}
-	if l.Sort != "" {
-		m["sort"] = l.Sort
-	}
-
-	return m
+// GetACLEntries returns a ListPaginator for paginating through the resources.
+func (c *Client) GetACLEntries(i *ListACLEntriesInput) *ListPaginator[ACLEntry] {
+	return NewPaginator[ACLEntry](c, &ListInput{
+		Direction: i.Direction,
+		Sort:      i.Sort,
+		Page:      i.Page,
+		PerPage:   i.PerPage,
+	}, ACLEntriesPath(i.ServiceID, i.ACLID))
```

Comment:
It feels a little odd for `GetACLEntries` to take `ListACLEntriesInput` -- did you consider splitting these into separate types?

My initial thought was that you could have:

```go
type ListInput struct {
	Direction string
	Page int
	PerPage int
	Sort string
}

type ListACLEntriesInput {
	ACLID string
	ServiceID string
}

type GetACLEntriesInput {
	ListACLEntriesInput
	ListInput
}
```

but I think this doesn't quite work because `Direction` and `Sort` are relevant to `ListACLEntriesInput` too.

So I don't have a recommendation here, really.  I think you could probably construct a bunch of different types and compose them together here but I doubt it's worthwhile.  Curious to hear your thoughts on this.

Reply:
> It feels a little odd for GetACLEntries to take ListACLEntriesInput 

Yup. Agreed. I was going for the easy path of being able to pass the input for the List input directly into the Get.

I'll have a think about how to do this better.

Reply:
OK I opted for a small bit of duplication to give the GetN methods their own input.